GitUp/Application/AppDelegate.m

```diff
@@ -68,6 +79,113 @@ - (BOOL)canBecomeKeyWindow {
 
 @end
 
+@interface GIWelcomeWindow__DraggingInfoExtractor : NSObject
+- (BOOL)canDragItem:(id<NSDraggingInfo>)sender;
+- (NSArray <NSURL *>*)draggingItems:(id<NSDraggingInfo>)sender;
+@end
+
+@implementation GIWelcomeWindow__DraggingInfoExtractor
+- (BOOL)canDragItem:(id<NSDraggingInfo>)sender {
+  return [self draggingItems:sender].count > 0;
+}
+- (NSArray <NSURL *>*)draggingItems:(id<NSDraggingInfo>)sender {
+  return [[sender draggingPasteboard] readObjectsForClasses:@[NSURL.class] options:nil];
+}
+@end
+
+@interface GIWelcomeWindow__DestinationView : NSView
+@property (weak, nonatomic, readwrite) AppDelegate *appDelegate;
+@property (strong, nonatomic, readwrite) GIWelcomeWindow__DraggingInfoExtractor *draggingExtractor;
+@property (assign, nonatomic, readwrite) BOOL receivingDrag;
+@property (strong, nonatomic, readwrite) NSView *imageView;
+@end
+
+@interface GIWelcomeWindow__DestinationView (Configurations)
+- (instancetype)configuredWithAppDelegate:(AppDelegate *)appDelegate;
```

Comment:
No point to this category and it's bad if anything for the dyld cache. Also see other comment about not injecting `AppDelegate`.